src/codecs/bmp/decoder.rs

```diff
@@ -295,17 +295,10 @@ fn extend_buffer(buffer: &mut Vec<u8>, full_size: usize, blank: bool) -> &mut [u
         old.copy_from_slice(&new[..old_size]);
         new
     } else {
```

Comment:
I think we can get rid of this part now, and just use `copy_within` unconditionally?

Reply:
I kept them separate because the >=2len path's `copy_from_slice` uses memcpy rather than memmove like `copy_within`, but it seems most libc impls will optimize non-overlapping memmove to a memcpy anyway so that seems like a decent simplification